libiocage/lib/Release.py

```diff
@@ -310,75 +369,129 @@ def update(self):
         dataset.snapshot(snapshot_name, recursive=True)
 
         jail = libiocage.lib.Jail.Jail({
-            "uuid"              : str(uuid.uuid4()),
-            "basejail"          : False,
-            "allow_mount_nullfs": "1",
-            "release"           : self.name,
-            "securelevel"       : "0"
-        },
+                "uuid"              : str(uuid.uuid4()),
+                "basejail"          : False,
+                "allow_mount_nullfs": "1",
+                "release"           : self.name,
+                "securelevel"       : "0"
+            },
+            new=True,
             logger=self.logger,
             zfs=self.zfs,
             host=self.host
         )
 
         jail.dataset_name = self.dataset_name
 
+        changed = False
+        try:
+            if self.host.distribution.name == "HardenedBSD":
+                changed = self._update_hbsd_jail(jail)
+            else:
+                changed = self._update_freebsd_jail(jail)
+        except:
+            self.logger.verbose(
+                "There was an error updating the Jail - reverting the changes"
+            )
+            jail.stop(force=True)
+            self.zfs.get_snapshot(snapshot_name).rollback(force=True)
+            raise
```

Comment:
Perhaps we should have a class here?

Reply:
The whole try/except block here is only there to ensure proper teardown in case of a failure. This means when anything goes wrong, we stop the helper jail and roll back the dataset. So the kind of exception that occurred does not matter here.